### PR TITLE
refactor: Added docstrings to BackgroundCommand(#1083)

### LINKED ISSUE #1083: BackgroundCommand/Process should be better documented
**What problem or use case are you trying to solve?**

The BackgroundCommand/Process class is not very well documented, despite the logic being somewhat complex. Docstrings should be added: https://github.com/OpenDevin/OpenDevin/blob/d27895d018098860e5c3d6613d0cf1b4a66a8478/opendevin/sandbox/sandbox.py#L15

### PR DESCRIPTION
**Pull request by JUSTSUJAY**
Added docstring to BackgroundCommand in opendevin/sandbox/sandbox.py
closes #1083

### CODE REVIEW

**Comment by neubig** [comment]
Thanks @JUSTSUJAY ! But could you please add a bit more detail about what actually goes on in the functions? All of the information in the current docstrring is stuff that is not very hard to deduce from just the function and argument names, but I think this would be more informative if it went into a bit more depth.

Also, linting seems to be failing, could you please fix the linting errors? Thank you!

**Comment by JUSTSUJAY** [comment]
Hey @neubig I have elaborated on how the functions work in my recent change, do you want me to add anything more?
and while going through the files, I noticed that the `result` attribute of **BackgroundCommand** class is being passed instance of `preprocess.Popen` class and in this file you can see that in `read_logs()` function, we are accessing **result.output**
but as I checked there's no output attribute for Popen, it's stdout

can you just check on it once too?

File: opendevin/sandbox/sandbox.py [hunk 1]
```diff
@@ -6,13 +6,56 @@
...
+            pid (int): The process ID (PID) of the command.
+        """
         self.id = id
         self.command = command
         self.result = result
         self.pid = pid
 
     def parse_docker_exec_output(self, logs: bytes) -> Tuple[bytes, bytes]:
+        """
+        Explaination:
```
Comment (neubig) [inline opendevin/sandbox/sandbox.py:30]: I don't think this "Explanation" is necessary, so make delete it. But if you'd like to keep it, please fix the typo
```suggestion
```

**Review by neubig — COMMENTED** [review]
Looks good with just one small comment

**Comment by JUSTSUJAY** [comment]
Fixed it

**Review by neubig — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**